core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -170,6 +171,36 @@ public static boolean hasPlugin(MavenProject project, String plugin) {
         return project.getPlugin(plugin) != null;
     }
 
+    public static boolean hasPluginOfAnyGroupId(MavenProject project, String pluginArtifact) {
+        return getPluginOfAnyGroupId(project, pluginArtifact) != null;
+    }
+
+    public static Plugin getPluginOfAnyGroupId(MavenProject project, String pluginArtifact) {
+        return getPlugin(project, null, pluginArtifact);
+    }
+
+    /**
+     * Returns the plugin with the given groupId (if present) and artifactId (if present).
+     */
+    public static Plugin getPlugin(MavenProject project, String groupId, String artifactId) {
+        if (groupId == null && artifactId == null) {
+            throw new IllegalArgumentException("groupId and artifactId cannot be both null");
+        }
+
+        List<Plugin> plugins = project.getBuildPlugins();
+        if (plugins != null) {
+            for (Plugin plugin : plugins) {
+                boolean matchesArtifactId = artifactId == null || Objects.equal(artifactId, plugin.getArtifactId());
```

Comment:
In which case makes it sense to have `null` artifactId ? The method would return the first pugin for a given group then.

Reply:
It doesn't make a lot of sense, as it returns a single result at most.